src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -34,6 +41,22 @@ internal VisualStudioWorkspace(HostServices hostServices, WorkspaceBackgroundWor
             }
         }
 
+        private void OnCacheFlushRequested(object sender, EventArgs e)
+        {
+            if (_backgroundCompiler != null)
+            {
+                _backgroundCompiler.Dispose();
+                _backgroundCompiler = null; // PartialSemanticsEnabled will now return false
```

Comment:
I'm curious, @pharring, have you tested what the VS experience is like when we do turn off partial semantics and have no background compiler? I'd be curious to know what the "large solution" experience is in this case...